app/components/submission/WithCurrentSubmission.js

```diff
@@ -1,110 +1,69 @@
-import gql from 'graphql-tag'
+import { cloneDeep } from 'lodash'
 import { withApollo } from 'react-apollo'
 import React from 'react'
 import omitDeep from 'omit-deep-lodash'
 import ErrorPage from '../global/ErrorPage'
 import { MANUSCRIPTS_QUERY } from '../dashboard/DashboardPage'
-
-const manuscriptFragment = gql`
-  fragment WholeManuscript on Manuscript {
-    id
-    title
-    manuscriptType
-    subjectAreas
-    suggestedSeniorEditors
-    opposedSeniorEditors {
-      name
-      reason
-    }
-    suggestedReviewingEditors
-    opposedReviewingEditors {
-      name
-      reason
-    }
-    suggestedReviewers {
-      name
-      email
-    }
-    opposedReviewers {
-      name
-      email
-      reason
-    }
-    noConflictOfInterest
-    files {
-      name
-      type
+import {
+  GET_CURRENT_SUBMISSION,
+  CREATE_SUBMISSION,
+  UPDATE_SUBMISSION,
+  FINISH_SUBMISSION,
+} from './Mutations'
+
+/*
+ * Data Modifiers -
+ *    CosubmissionModifier
+ *
+ * These classes provide a common interface for the main class
+ * 'WithCurrentSubmission' to delgate transforming the data model to the
+ * view model.
+ */
+class CosubmissionModifier {
```

Comment:
The data modifier concept is good. But given we don't use classes anywhere else in the codebase (except when forced to do so on some React components) I would avoid them here too for consistency. A plain object with three function properties would do the trick.

Reply:
Yeah - this is an interesting one. Without sparking a huge conversation - I am very much pro-class sort of a guy :) ... and certainly when looking at other react code - including PubSweet and other xpub apps, I would like to go the same way.... But I get your point!